src/deno.h

```diff
@@ -37,11 +40,17 @@ int deno_execute(Deno* d, const char* js_filename, const char* js_source);
 // Routes message to the javascript callback set with deno.recv(). A false
 // return value indicates error. Check deno_last_exception() for exception text.
 // 0 = fail, 1 = success
+// After calling deno_send(), the caller no longer owns `buf` and must not use
+// it; deno_send() is responsible for releasing it's memory.
+// TODO(piscisaureus) In C++ and/or Rust, use a smart pointer or similar to
+// enforce this rule.
 int deno_send(Deno* d, deno_buf buf);
 
 // Call this inside a deno_recv_cb to respond synchronously to messages.
 // If this is not called during the life time of a deno_recv_cb callback
 // the deno.send() call in javascript will return null.
+// After calling deno_set_response(), the caller no longer owns `buf` and must
+// not access it; deno_set_response() is responsible for releasing it's memory.
 void deno_set_response(Deno* d, deno_buf buf);
```

Comment:
If these are to be the semantics, why not change the signature of `deno_recv_cb` to instead have users return the buf and completely remove `deno_set_response`:
```c
typedef deno_buf (*deno_recv_cb)(Deno* d, deno_buf buf);
```

Reply:
I had something else in mind (deno_set_response and deno_send are going to merge)

Reply:
Does that mean that deno.send in JS will not return values ? Or that deno_send during recv_cb has special semantics that sets the return value ?

Reply:
deno.send will return values, or something like that.
(maybe deno.send won't, but then there'll be another method e.g. deno.sync() or deno.transact() that does).